src/Microsoft.ML.Api/SchemaDefinition.cs

```diff
@@ -332,8 +332,18 @@ public static SchemaDefinition Create(Type userType)
 
                 if (fieldInfo.GetCustomAttribute<NoColumnAttribute>() != null)
                     continue;
-                var mappingAttr = fieldInfo.GetCustomAttribute<ColumnNameAttribute>();
-                var name = mappingAttr == null ? fieldInfo.Name : (mappingAttr.Name ?? fieldInfo.Name);
+                var mappingAttr = fieldInfo.GetCustomAttribute<ColumnAttribute>();
+                string attrName = null;
+                if (mappingAttr != null)
+                {
+                    attrName = mappingAttr.Name;
+                }
+                else
+                { 
+                    var mappingNameAttr = fieldInfo.GetCustomAttribute<ColumnNameAttribute>();
+                    attrName = mappingNameAttr != null ? mappingNameAttr.Name : null;
+                }
+                var name = attrName == null ? fieldInfo.Name : attrName;
                 // Disallow duplicate names, because the field enumeration order is not actually
```

Comment:
Actually wait... if you just cut out these 12 lines of code and replaced it with these:

```csharp
var mappingAttr = fieldInfo.GetCustomAttribute<ColumnAttribute>();
var mappingNameAttr = fieldInfo.GetCustomAttribute<ColumnNameAttribute>();
string name = mappingAttr?.Name ?? mappingNameAttr?.Name ?? fieldInfo.Name;
```

you could make this much shorter, and much, much clearer. The `??` and `?.` operators are fantastic aids for this sort of "if null/not null" logic...